manifest/autoversion/autoversion.go

```diff
@@ -80,6 +86,44 @@ blocks:
 
 	// Update the manifest and write it out to disk.
 	content, err = hcl.MarshalAST(ast)
+
+	// prepare for PopulateDigestsForVersion
+	var absolute string
+	absolute, err = filepath.Abs(path)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	dir := filepath.Dir(absolute)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	annotated := &hmanifest.AnnotatedManifest{
+		FS:   os.DirFS(dir),
+		Name: strings.Replace(filepath.Base(path), ".hcl", "", 1),
+		Path: absolute,
+	}
+
+	annotated, err = hmanifest.LoadManifestBytes(content, annotated)
```

Comment:
This shouldn't be serialising, deserialising, then serialising again all in here. That's a sign the abstractions aren't correct. At the very least, this should be moved out into a `LoadManifestFromAST()` or similar.